scheduler/src/cook/rest/api.clj

```diff
@@ -1639,6 +1652,38 @@
                        (comp first))]
     (base-read-instances-handler conn is-authorized-fn {:handle-ok handle-ok})))
 
+(defn update-instance-progress-handler
+  [conn is-authorized-fn leadership-atom leader-selector progress-aggregator-chan]
+  (base-cook-handler
+    {:allowed-methods [:post]
+     :service-available? (fn [ctx]
+                           ;; injecting ::instances into ctx for later handlers
+                           (let [instance-uuid (get-in ctx [:request :params :uuid])]
+                             [true {::instances [instance-uuid]}]))
+     :allowed? (partial instance-request-allowed? conn is-authorized-fn)
+     :exists? (constantly false)  ;; triggers path for moved-temporarily?
+     :existed? instance-request-exists?
+     :can-post-to-missing? (constantly false)
+     :moved-temporarily? (fn [ctx]
+                           ;; only the leader handles progress updates
+                           ;; the client is expected to cache the redirect location
+                           (let [request-path (get-in ctx [:request :uri])]
+                             (if @leadership-atom
+                               [false {}]
+                               [true {:location (str (leader-url leader-selector) request-path)}])))
```

Comment:
Nitpick: should we use a namespaced keyword here?

Reply:
`:location` is something that Liberator recognizes for doing redirects, not something I picked.